pkg/http/sse.go

```diff
@@ -0,0 +1,97 @@
+package http
+
+import (
+	"net/http"
+
+	"github.com/ThreeDotsLabs/watermill/message"
+	"github.com/go-chi/render"
+)
+
+type StreamAdapter interface {
+	// ResponseProvider returns the response to be sent back to client.
+	// Any errors that occur should be handled and written to `w`, returning false as `ok`.
+	ResponseProvider(w http.ResponseWriter, r *http.Request) (response interface{}, ok bool)
+	// Validator validates if the incoming message should be handled by this handler.
+	// Typically this involves checking some kind of model ID.
+	Validator(r *http.Request, msg *message.Message) (ok bool)
+}
+
+type SSEHandler struct {
+	subscriber    message.Subscriber
+	topic         string
+	streamAdapter StreamAdapter
+}
+
+func NewSSEHandler(
+	subscriber message.Subscriber,
+	topic string,
+	streamAdapter StreamAdapter,
+) SSEHandler {
+	return SSEHandler{
```

Comment:
we could probably use some parameter validation

Reply:
Added in 265aeb18900b0283175d746a452b6334f3f575d3